orttraining/orttraining/training_ops/cpu/nn/layer_norm.cc

```diff
@@ -11,34 +11,25 @@ namespace contrib {
-      T,                                                          \
-      kCpuExecutionProvider,                                      \
-      KernelDefBuilder()                                          \
-          .TypeConstraint("T", DataTypeImpl::GetTensorType<T>()), \
-      LayerNormGrad<T, true>);                                    \
-  ONNX_OPERATOR_TYPED_KERNEL_EX(                                  \
-      InvertibleLayerNormalizationGrad,                           \
-      kMSDomain,                                                  \
-      1,                                                          \
-      T,                                                          \
-      kCpuExecutionProvider,                                      \
-      KernelDefBuilder()                                          \
-          .TypeConstraint("T", DataTypeImpl::GetTensorType<T>()), \
-      InvertibleLayerNormGrad<T>);
+#define REGISTER_KERNEL_TYPED(T)                                                                          \
+  ONNX_OPERATOR_TYPED_KERNEL_EX(LayerNormalizationGrad, kMSDomain, 1, T, kCpuExecutionProvider,           \
+                                KernelDefBuilder()                                                        \
+                                    .TypeConstraint("T", DataTypeImpl::GetTensorType<T>())                \
+                                    .TypeConstraint("U", DataTypeImpl::GetTensorType<T>())                \
+                                    .TypeConstraint("V", DataTypeImpl::GetTensorType<T>()),               \
+                                LayerNormGrad<T, false>);                                                 \
+  ONNX_OPERATOR_TYPED_KERNEL_EX(SimplifiedLayerNormalizationGrad, kMSDomain, 1, T, kCpuExecutionProvider, \
+                                KernelDefBuilder()                                                        \
+                                    .TypeConstraint("T", DataTypeImpl::GetTensorType<T>())                \
+                                    .TypeConstraint("U", DataTypeImpl::GetTensorType<T>())                \
+                                    .TypeConstraint("V", DataTypeImpl::GetTensorType<T>()),               \
+                                LayerNormGrad<T, true>);                                                  \
+  ONNX_OPERATOR_TYPED_KERNEL_EX(InvertibleLayerNormalizationGrad, kMSDomain, 1, T, kCpuExecutionProvider, \
+                                KernelDefBuilder()                                                        \
+                                    .TypeConstraint("T", DataTypeImpl::GetTensorType<T>())                \
+                                    .TypeConstraint("U", DataTypeImpl::GetTensorType<T>())                \
+                                    .TypeConstraint("V", DataTypeImpl::GetTensorType<T>()),               \
+                                InvertibleLayerNormGrad<T>);
 
```

Comment:
We missed to update the new types in this files. 

    template <typename T, bool simplified>
    LayerNormGrad<T, simplified>::LayerNormGrad(const OpKernelInfo& op_kernel_info)

Reply:
For cpu kernel, all inputs and outputs are in T. The class is for kernel implementation, only T is needed. During kernel registry, we will map all T, U, V to the given type we want to register.